akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -691,6 +690,29 @@ object Source {
     new Source(scaladsl.Source.zipWithN[T, O](seq => zipper.apply(seq.asJava))(seq))
   }
 
+  /**
+   * Creates a `Source` that is materialized as an [[akka.stream.BoundedQueueSource]].
+   * You can push elements to the queue and they will be emitted to the stream if there is demand from downstream,
+   * otherwise they will be buffered until request for demand is received. The buffer size is passed in as a parameter.
+   * Elements in the buffer will be discarded if downstream is terminated.
+   *
+   * Pushed elements may be dropped if there is no space available in the buffer. Elements will also be dropped if the
+   * queue is completed or failed or if the stream completes.
+   *
+   * Acknowledgement of pushed elements is immediate.
+   * [[akka.stream.BoundedQueueSource.offer]] returns [[akka.stream.QueueOfferResult]] which is implemented as:
+   *
+   * `QueueOfferResult.enqueued()`     element was added to buffer
+   * `QueueOfferResult.dropped()`      element was dropped
+   * `QueueOfferResult.QueueClosed`    the queue was completed with [[akka.stream.BoundedQueueSource.complete]]
+   * `QueueOfferResult.Failure`        the queue was failed with [[akka.stream.BoundedQueueSource.fail]] or if the
+   *                                   stream failed
+   *
+   * @param bufferSize size of buffer in element count
```

Comment:
```suggestion
   * @param bufferSize size of the buffer in number of elements
```

or maybe just leave it away

Reply:
:+1: